pkgs/development/compilers/llvm/6/crtbegin-and-end.patch

```diff
@@ -0,0 +1,595 @@
+Get crtbegin and crtend without compiler GCC! PR is at https://reviews.llvm.org/D28791
```

Comment:
Someone noticed this:

http://lists.llvm.org/pipermail/llvm-commits/Week-of-Mon-20190513/652039.html

We probably want to wait for that to be resolved.

Reply:
Also, I think we can use fetchpatch here as well IIRC

Reply:
The patch in question is already being applied to LLVM 7 to make cross-compilation work, I merely backported it to 6 (I'm doing Haskell cross-compilation and GHC 8.6.4 needs LLVM 6.) Perhaps since this is already in place for LLVM 7 it could just be merged for LLVM 6 and then all the LLVMs could be updated with the resolution to the above thread?

Also, I'm cross-compiling to ARM and have not encountered the mentioned issue with the patched LLVM v6 in this PR.

Reply:
You shouldn't need clang for GHC cross compilation. LLVM should just work on its own. Are you on nixpkgs-unstable or 19.03?

Reply:
I'm using unstable, but I'm using IOHK's haskell nix infrastructure. I'm attempting to use clang as the compiler for c files included with Haskell libraries, trying to work around some issues with iserv and template Haskell during cross-compilation.

Reply:
On the other hand, it took longer into the build than I expected, but that error from the mailing list did now show up. So I guess this isn't ready to be merged yet.